lib/slack.py

```diff
@@ -115,16 +122,34 @@ def suggest_video(self, user_info, team_info, channel_info, query="", action_inf
             self._mysql.execute("select videos from video_suggestions where search_id = {}".format(search_id))
             videos = literal_eval(self._mysql.fetchone()[0])
             num_vids = len(videos)
-            suggested_video = self._youtube.get_video_metadata(videos[randint(0, num_vids) % num_vids])
+            index = randint(0, num_vids) % num_vids
+            suggested_video = self._youtube.get_video_metadata(videos[index])
             suggested_video['id'] = { "videoId": suggested_video['id'] }
             message_to_send = self.__build_message__(suggested_video, channel=channel,
-                                                       username=username, search_id=search_id)
+                                                       username=username, search_id=search_id, index=index)
             message_to_send['replace_original'] = True
             return message_to_send
+        elif 'name' in action_info.keys() and action_info['name'] == 'next':
```

Comment:
Some of the code here is repetitive in the shuffle button's if block, probably worth making that its own function